### PR TITLE
Restore running xacts from CLOG on replica startup, take two

### PR DESCRIPTION
**Pull request by hlinnaka**
This is same as PR #7288, with some changes:

- I moved the RestoreRunningXactsFromClog() call from vendor/postgres repositories to the neon extension. It's now called from neon_rmgr's rm_startup routine.
- I'm now constructing a RunningTransactions snapshot and using ProcArrayApplyRecoveryInfo() to install it.
- If there are too many in-progress XIDs in the CLOG, we now fall back to waiting for a running-xacts record, instead of risking incorrect query results.
- Rebased over current 'main'

@knizhnik: What do you think? If you agree with this general direction, I'd like to force-push this to PR #7288, and close this PR. I just didn't want to do that without asking first.

### CODE REVIEW

**Comment by hlinnaka** [comment]
The tests that this adds are failing, that's expected. They demonstrate the two cases where this optimization fails: in one case the known-assigned xids area is overflown so we wait for the running-xacts record, and in the other case it's overflown later, in WAL replay, so we crash with `FATAL:  too many KnownAssignedXids`. We should adjust the test cases so that they expect those results, and pass.

File: pgxn/neon_rmgr/neon_rmgr.c [hunk 1]
```diff
@@ -81,10 +81,145 @@ neon_rm_redo(XLogReaderState *record)
...
+
+	pfree(xids);
+}
+
 static void
 neon_rm_startup(void)
 {
-	/* nothing to do here */
+	if (standbyState == STANDBY_INITIALIZED)
+		RestoreRunningXactsFromClog();
```
Comment (knizhnik) [inline pgxn/neon_rmgr/neon_rmgr.c:220]: I still not sure that neon_rm_startup is right place to restore running xacts.
Is it called before or after `StartupXLOG` ?

**Review by knizhnik — APPROVED** [review]
If

**Comment by hlinnaka** [comment]
Per discussion with Konstantin, I force-pushed this to PR #7288. Let's continue there.

**Pull request closed without merging**